Documentation/project-docs/glossary.md

```diff
@@ -1,92 +1,224 @@
+* **IL**. Intermediate language. Higher level .NET languages, such as C#,
+  compile down to a hardware agnostic instruction set, which is called
+  Intermediate Language (IL). IL is sometimes referred to as MSIL (Microsoft IL)
+  or CIL (Common IL).
+
+* **GC**. Garbage collector. Garbage collection is an implementation of
+  automatic memory management. .NET Framework and .NET Core currently uses a
+  generational garbage collector, i.e. it groups objects into generations to
+  limit the number of nodes it has to walk for determining which objects are
+  alive. This speeds up collection times.
+
+* **JIT**. Just in time compiler. This technology compiles IL to machine code
+  that the processor understands. It's called JIT because compilation happens on
+  demand and is performed on the same machine the code needs to run on. Since
+  JIT compilation occurs during execution of the application, compile time is
+  part of the run time. Thus, JIT compilers have to trade spending more time
+  optimizing code with the savings the resulting code can produce. But a JIT
+  knows the actual hardware and can free developers from having to ship
+  different implementations. For instance, our vector library relies on the JIT
+  to use the highest available SIMD instruction set.
+
+* **AOT**. Ahead of time compiler. Similar to JIT, this compiler also translates
+  IL to machine code. In contrast to JIT compilation, AOT compilation happens
+  before the application is executed and is usually performed on a different
+  machine. AOT tool chains don't trade runtime for compile time and thus can
+  spend more time optimizing. Since the context of AOT is the entire
+  application, the AOT compiler can also perform cross module linking and whole
+  program analysis, which means that all references are followed and a single
+  executable is produced.
+
+* **NGEN**. Native (image) generation. You can think of this technology as a
+  persistent JIT compiler. It usually compiles code on the machine where the
+  code will be executed, but compilation typically occurs at install time.
+
+* **CoreFX**. Core framework. Conceptually a set of `System.*` (and to a limited
+  extent `Microsoft.*`) libraries that make up the lower layer of the .NET
+  library stack. It's what most people would think of as the Base Class Library
+  (BCL). The BCL is a general purpose, lower level set of functionality that
+  higher-level frameworks, such as WCF and ASP.NET, build on. The source code of
+  the .NET Core library stack is contained in the [CoreFX repo][corefx].
+  However, the majority of the .NET Core APIs are also available in the .NET
+  Framework, so you can think of CoreFX as a fork of the .NET Framework library
+  stack.
+
+* **CLI** Command Line Interface --or-- Common Language Runtime.
```

Comment:
CLI in this context refers to the .NET Core CLI specifically presumably not a generic command line.

I have not heard anyone say CLI in the 2nd sense in some time especially since it's ambiguous now.

Reply:
I could also see it being used to talk about making CLI apps.